contracts/src/Plaza.sol

```diff
@@ -126,18 +123,33 @@ contract Plaza is ERC20, Ownable, ReentrancyGuard {
 
     function withdrawFunds() external onlyOwner nonReentrant {
         require(targetAmount > 0, "Project does not accept funds");
```

Comment:
We don't need this require here, it is sufficient to have it in line 98.

Moreover, having this require here in line 125 is dangerous, because people could inadvertently transfer funds to this contract (not necessarily using the `contribute` fund) and then these funds would become irrecoverable.

So: 

* Remove the require statement from line 125.
* Implement receive and fallback functions to revert if users inadvertently try to transfer funds to the contract by means other than the `contribute` function.